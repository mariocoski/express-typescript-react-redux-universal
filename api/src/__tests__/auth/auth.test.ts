describe('AUTH', () => {

  const request = require('supertest');
  let app;

  beforeEach(() => {
    app = require('../../server');
  });

  afterEach(() => {
    app.close();
  });

  it('should fail to create a user without input', async () => {
    expect.assertions(2);
    const response = await request(app).post('/auth/register');
    expect(response.statusCode).toBe(422);
    expect(response.text).toMatchSnapshot();
  });

  it('should fail to create a user when email is invalid', async () => {
    expect.assertions(2);
    const response = await request(app).post('/auth/register')
    .type('form')
    .send({
      email: 'invalidEmail'
    });
    expect(response.statusCode).toBe(422);
    expect(response.text).toMatchSnapshot();
  });

  //   .post('register', register)
    //   .post('login', requireLogin, login)
    //   .post('forgot-password', forgotPassword)
    //   .post('reset-password/:token', verifyToken)
    //   .post('me-from-token', meFromToken);


});