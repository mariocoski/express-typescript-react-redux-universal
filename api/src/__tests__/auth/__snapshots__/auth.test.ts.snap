// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTH should fail to create a user when email is invalid 1`] = `"{\\"errors\\":{\\"email\\":{\\"location\\":\\"body\\",\\"param\\":\\"email\\",\\"msg\\":\\"Email is required\\"},\\"password\\":{\\"location\\":\\"body\\",\\"param\\":\\"password\\",\\"msg\\":\\"passwords must be at least 5 chars long and contain one number\\"}}}"`;

exports[`AUTH should fail to create a user without input 1`] = `"{\\"message\\":\\"Email is required\\"}"`;
